### PR TITLE
HBASE-21674 Port HBASE-21652 (Refactor ThriftServer making thrift2 server inherited from thrift1 server) to branch-1

### PR DESCRIPTION
**Pull request by YutSean**
https://issues.apache.org/jira/browse/HBASE-21674
Some DDL operations of Thrift2 are not implemented. These will be implemented in the future.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   7m 21s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  2s |  No case conflicting files found.  |
| +0 :ok: |  jshint  |   0m  0s |  jshint was not available.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 9 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +0 :ok: |  mvndep  |   2m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   8m 16s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 49s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   1m  4s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 12s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m 35s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   1m 54s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   1m 51s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m 14s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 19s |  Maven dependency ordering for patch  |
| -1 :x: |  mvninstall  |   2m  6s |  root in the patch failed.  |
| +1 :green_heart: |  compile  |   0m 53s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 53s |  the patch passed  |
| -1 :x: |  compile  |   0m 37s |  hbase-thrift in the patch failed with JDK Azul Systems, Inc.-1.7.0_272-b10.  |
| -1 :x: |  javac  |   0m 37s |  hbase-thrift in the patch failed with JDK Azul Systems, Inc.-1.7.0_272-b10.  |
| -1 :x: |  checkstyle  |   0m 32s |  hbase-common: The patch generated 1 new + 0 unchanged - 0 fixed = 1 total (was 0)  |
| -1 :x: |  checkstyle  |   0m 45s |  hbase-thrift: The patch generated 354 new + 83 unchanged - 76 fixed = 437 total (was 159)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 23s |  patch has no errors when building our shaded downstream artifacts.  |
| -1 :x: |  hadoopcheck  |   1m 35s |  The patch causes 16 errors with Hadoop v2.8.5.  |
| -1 :x: |  hadoopcheck  |   4m  8s |  The patch causes 16 errors with Hadoop v2.9.2.  |
| -1 :x: |  javadoc  |   0m 34s |  hbase-thrift-jdkAzulSystems,Inc.-1.8.0_262-b19 with JDK Azul Systems, Inc.-1.8.0_262-b19 generated 13 new + 0 unchanged - 0 fixed = 13 total (was 0)  |
| -1 :x: |  javadoc  |   3m 20s |  hbase-thrift-jdkAzulSystems,Inc.-1.7.0_272-b10 with JDK Azul Systems, Inc.-1.7.0_272-b10 generated 13 new + 0 unchanged - 0 fixed = 13 total (was 0)  |
| -1 :x: |  findbugs  |   0m 38s |  hbase-thrift in the patch failed.  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 14s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  |   0m 39s |  hbase-thrift in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m 26s |  The patch does not generate ASF License warnings.  |
|  |   |  59m  2s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2936 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile jshint |
| uname | Linux 7c80bdb29a20 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2936/out/precommit/personality/provided.sh |
| git revision | branch-1 / 7d80d99 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
| mvninstall | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/patch-mvninstall-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/patch-compile-hbase-thrift-jdkAzulSystems,Inc.-1.7.0_272-b10.txt |
| javac | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/patch-compile-hbase-thrift-jdkAzulSystems,Inc.-1.7.0_272-b10.txt |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/diff-checkstyle-hbase-common.txt |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/diff-checkstyle-hbase-thrift.txt |
| hadoopcheck | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/patch-javac-2.8.5.txt |
| hadoopcheck | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/patch-javac-2.9.2.txt |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/diff-javadoc-javadoc-hbase-thrift-jdkAzulSystems,Inc.-1.8.0_262-b19.txt |
| javadoc | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/diff-javadoc-javadoc-hbase-thrift-jdkAzulSystems,Inc.-1.7.0_272-b10.txt |
| findbugs | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/patch-findbugs-hbase-thrift.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/artifact/out/patch-unit-hbase-thrift.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/testReport/ |
| Max. process+thread count | 181 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-thrift U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2936/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/AlreadyExists.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
  * An AlreadyExists exceptions signals that a table with the specified
  * name already exists
  */
-@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2020-04-16")
+@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2021-01-12")
```
Comment (aimiebell) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/AlreadyExists.java:14]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/IOError.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
  * to the Hbase master or an Hbase region server.  Also used to return
  * more general Hbase error conditions.
  */
-@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2020-04-16")
+@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2021-01-12")
```
Comment (aimiebell) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/IOError.java:15]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/IllegalArgument.java [hunk 3]
```diff
@@ -11,7 +11,7 @@
  * An IllegalArgument exception indicates an illegal or invalid
  * argument was passed into a procedure.
  */
-@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2020-04-16")
+@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2021-01-12")
```
Comment (aimiebell) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/IllegalArgument.java:14]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/TAppend.java [hunk 4]
```diff
@@ -10,7 +10,7 @@
 /**
  * An Append object is used to specify the parameters for performing the append operation.
  */
-@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2020-04-16")
+@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2021-01-12")
```
Comment (aimiebell) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/TAppend.java:13]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/TCell.java [hunk 5]
```diff
@@ -13,7 +13,7 @@
  * the timestamp of a cell to a first-class value, making it easy to take
  * note of temporal data. Cell is used all the way from HStore up to HTable.
  */
-@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2020-04-16")
+@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2021-01-12")
```
Comment (aimiebell) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/TCell.java:16]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/TIncrement.java [hunk 6]
```diff
@@ -11,7 +11,7 @@
  * For increments that are not incrementColumnValue
  * equivalents.
  */
-@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2020-04-16")
+@javax.annotation.Generated(value = "Autogenerated by Thrift Compiler (0.13.0)", date = "2021-01-12")
```
Comment (aimiebell) [inline hbase-thrift/src/main/java/org/apache/hadoop/hbase/thrift/generated/TIncrement.java:14]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

**Pull request closed without merging**